zerver/forms.py

```diff
@@ -208,6 +216,20 @@ def clean_email(self) -> str:
         if realm.is_zephyr_mirror_realm:
             email_is_not_mit_mailing_list(email)
 
+        if settings.BILLING_ENABLED:
+            try:
+                check_license_limit(realm)
+            except LicenseLimitError:
+                message = f"Signup of {email} failed because of license limit. Please increase the licenses."
+                send_message_to_signup_notification_stream(
+                    get_system_bot(settings.NOTIFICATION_BOT), realm, message
+                )
+                raise ValidationError(
+                    _(
+                        "The organization is using all it's licenses and can't accept new members. Please contact the organization admin."
+                    )
+                )
```

Comment:
I think adding these checks in HomepageForm is not quite enough? Invites will not be sendable, but it looks like arbitrary numbers of users will be able to join via pre-generated invite links - because `accounts_register` (`/accounts/register/` endpoint) doesn't verify.

Reply:
Yeah. Right. We need to add a check there as well.